File: project 6/ruby/task 1&2 codes/task2a.rb
=begin
*	Mkhanyisi Gamedze
*	CS333 Programming languages
*	10 November 2017
*	Ruby : Functions and File I/O
=end


# function one
def add(a,b) 
   sum=a+b
   puts "#{a} + #{b} = #{sum}"
end

def subtract(a,b)
	return (a-b)
end
x=15
y=4
puts "addition"
add(x,y)


puts "difference"
diff=subtract(x,y)
puts "#{x} - #{y} = #{diff}"

File: project 6/C++  /Task 1& 2 Codes/task2a.cpp
/*
*	Mkhanyisi Gamedze
*	CS333 Programming languages
*	10 November 2017
*	C++: Functions and File I/O
*/

// basic file operations
#include <iostream> 
#include <fstream> // to both read & write to files
using namespace std;

int retInt(){
	cout<<"Int return statement"<<endl;
	return 1;
}
void retVoid(){
	cout<<"void return statement"<<endl;
}

/* demonstrating how functions can have void type  */ 
int main () {
	
	retInt();
	retVoid();
	
	
  return 0;  // main function cannot compile without a return statement
}
